src/libraries/System.Private.CoreLib/src/System/IO/StreamWriter.cs

```diff
@@ -320,11 +323,55 @@ private void Flush(bool flushStream, bool flushEncoder)
                     (_byteBuffer = new byte[_encoding.GetMaxByteCount(_charBuffer.Length)]);
             }
 
-            int count = _encoder.GetBytes(new ReadOnlySpan<char>(_charBuffer, 0, _charPos), byteBuffer, flushEncoder);
-            _charPos = 0;
-            if (count > 0)
+            if (OperatingSystem.IsIOS() || OperatingSystem.IsTvOS() || OperatingSystem.IsMacCatalyst())
             {
-                _stream.Write(byteBuffer.Slice(0, count));
+                for (int i = _lineStartPos; i < _charPos; i++)
+                {
+                    if (_charBuffer[i] == '\n')
+                    {
+                        int count = _encoder.GetBytes(new ReadOnlySpan<char>(_charBuffer, _lineStartPos, i - _lineStartPos), byteBuffer, flushEncoder);
+                        if (count > 0)
+                        {
+                            _acc.AddRange(byteBuffer.Slice(0, count).ToArray());
+                        }
+
+                        _stream.Write(_acc.ToArray().AsSpan<byte>());
```

Comment:
nit: avoid ToArray()
```suggestion
                        _stream.Write(CollectionsMarshal.AsSpan(_acc));
```